### PR TITLE
Add proof-producer-debug-gcc check in CI

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

File: .github/workflows/gcc-debug-proof-producer-nix-check.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      # https://github.com/actions/checkout/issues/1552
+      - name: Clean up after previous checkout
+        run: chmod +w -R ${GITHUB_WORKSPACE}; rm -rf ${GITHUB_WORKSPACE}/*;
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Run checks for debug proof-producer
```
Comment (oclaw) [inline .github/workflows/gcc-debug-proof-producer-nix-check.yml:20]: Maybe highlight somehow that no actual tests are running here, just basic compiled binary check?

Comment (AndreyMlashkin) [reply .github/workflows/gcc-debug-proof-producer-nix-check.yml:20]: True, thanks for noticing

Comment (AndreyMlashkin) [reply .github/workflows/gcc-debug-proof-producer-nix-check.yml]: corrected

**Review by oclaw — APPROVED** [review]

File: .github/workflows/gcc-debug-proof-producer-nix-check.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Build and Test on Linux Platforms
+
+on:
+  workflow_call:
+
+jobs:
+  build-and-test:
```
Comment (makxenov) [inline .github/workflows/gcc-debug-proof-producer-nix-check.yml:7]: Just a bit confusing name, we don't run any tests here

**Review by makxenov — APPROVED** [review]

**Review by oclaw — APPROVED** [review]

**Pull request merged**